### PR TITLE
Checkify: Fix closing over const Tracers in while_loop cond_f.

### PR DESCRIPTION
**Pull request by LenaMartens**

### CODE REVIEW

File: jax/_src/checkify.py [hunk 1]
```diff
@@ -862,10 +867,10 @@ def while_loop_error_check(error, enabled_errors, *in_flat, cond_nconsts,
   to_move = [False] * num_error_vals + [True] * cond_nconsts + [False] * len(carry)
   compat_cond_jaxpr = pe.move_binders_to_front(compat_cond_jaxpr_, to_move)
 
-  new_in_flat = [*c_consts, *b_consts, *err_vals, *carry]
+  new_in_flat = [*c_consts, *c_consts, *b_consts, *err_vals, *carry]
   all_out_vals = lax.while_p.bind(
       *new_in_flat, cond_nconsts=cond_nconsts, cond_jaxpr=compat_cond_jaxpr,
-      body_nconsts=body_nconsts, body_jaxpr=checked_body_jaxpr)
+      body_nconsts=body_nconsts+cond_nconsts, body_jaxpr=checked_body_jaxpr)
```
Comment (mattjj) [inline jax/_src/checkify.py:873]:
```suggestion
      body_nconsts=cond_nconsts+body_nconsts, body_jaxpr=checked_body_jaxpr)
```

That doesn't change the value of the expression but it reinforces that we're putting the original cond consts first.

**Review by mattjj — APPROVED** [review]
LGTM! Should we double check whether we are doing similar things in any other HOPs, like scan?

**Comment by LenaMartens** [comment]
I think `while_loop` is unique in that it constructs that special `body_fn` which also calls the `cond_fn`. None of the other HOPs do anything like that, but I'm happy to add some tests in a follow-up.

File: jax/_src/checkify.py (hunk 1 above)
Comment (LenaMartens) [reply jax/_src/checkify.py]: Done!

**Pull request merged**